### DISCUSSION TITLE
Need to be able to navigate through items using the keyboard in multiplayer or playlists

### DISCUSSION THREAD

**Discussion by Moltanicaa** [ppy/osu#18061] [Ideas] [answered]
At the moment it is not possible to navigate through the list of maps (relevant for playlists) using the arrows (left, right, up, down) on the keyboard. 
It is also not possible to switch between columns (to scroll the leaderboard, for example) using the Tab key.
The Enter key is also ignored if I want to play the selected map (because all focus is on the text field).

**Comment by frenzibyte** [comment]
Should be resolved in https://github.com/ppy/osu/pull/18090

**Comment by Moltanicaa** [comment]
I can see that moving through the objects is implemented, but it is still not possible to start the map using Enter. Should I start another thread for this?

**Reply by frenzibyte** [reply]
Yes please.

**Reply by Moltanicaa** [reply]
https://github.com/ppy/osu/discussions/18195

**Comment by Susko3** [comment]
This has regressed with https://github.com/ppy/osu/commit/b9590320b750d64ed532ce0621f54019940c1f4c. Up and down arrow keys now go trough the textbox history instead of the playlist items, with no way to unfocus the text box.

Maybe this is fine as this is a niche use case and there's a workaround: bind `Previous selection` and `Next selection` to something other than <kbd>&uarr;</kbd> <kbd>&darr;</kbd> (or the alphanumeric keys, as those are handled by the textbox when typing).

**Reply by frenzibyte** [reply]
This regression probably should be fixed, or at worst case have the feature explicitly removed. For me, I would be in favour of the chat text box not holding focus, and requiring to press <kbd>Tab</kbd> (or possibly pressing any letter key on the keyboard to make it automatically focus), and explicitly show that focus has been taken away from the playlist items control, by changing the colour of the item borders from yellow to grey-ish.

**Reply by frenzibyte** [reply, marked as answer]
https://github.com/ppy/osu/issues/25302
